app/Services/WidgetRegistry.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+namespace App\Services;
+
+use Illuminate\Contracts\Foundation\Application;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Log;
+use LibreNMS\Interfaces\Plugins\Hooks\DashboardWidgetHook;
+
+class WidgetRegistry
+{
+    /** @var Collection */
+    private $widgets;
+
+    /** @var Application */
+    private $app;
+
+    /** @var array core widgets part of LibreNMS */
+    private $coreWidgets = [
+        'alerts',
+        'alertlog',
+        'alertlog-stats',
+        'availability-map',
+        'component-status',
+        'custom-map',
+        'device-summary-horiz',
+        'device-summary-vert',
+        'device-types',
+        'eventlog',
+        'globe',
+        'generic-graph',
+        'graylog',
+        'generic-image',
+        'notes',
+        'server-stats',
+        'syslog',
+        'top-devices',
+        'top-errors',
+        'top-interfaces',
+        'worldmap',
+    ];
+
+    public function __construct(Application $app)
+    {
+        $this->app = $app;
+        $this->widgets = new Collection();
+        $this->registerCoreWidgets();
+        $this->registerPluginWidgets();
+    }
+
+    /**
+     * Register core widgets
+     */
+    private function registerCoreWidgets(): void
+    {
+        foreach ($this->coreWidgets as $widget) {
+            $this->widgets->put($widget, [
+                'name' => $widget,
+                'title' => trans("widgets.$widget.title"),
+                'controller' => $this->getControllerClass($widget),
+                'type' => 'core',
+                'metadata' => [],
+            ]);
+        }
+    }
+
+    /**
+     * Register widgets from plugins using service container tagged services
+     */
+    private function registerPluginWidgets(): void
+    {
+        try {
+            $taggedWidgets = $this->app->tagged('librenms.widget');
+
+            foreach ($taggedWidgets as $widget) {
+                if (! $widget instanceof DashboardWidgetHook) {
```

Comment:
You didn't get add DashboardWidgetHook.  Perhaps this should just be a regular interface instead of a "PluginHook".